Docs: define a default mapping for all indexes and all types using templates

fforbeck opened the pull request:
Documented how to define a default mapping to match all indexes and all types using custom template.
Useful when you have the name of the index and the type generated dynamically. With this approach you  can avoid sending a new PUT request for each new type/index you have created.

clintongormley commented:
Hi @fforbeck 

Thanks for the PR.  In general i like it, the only thing I don't like is the use of the `attachment` plugin.  We will be removing this plugin in the future (to be replaced by an attachment processor for the new ingest node functionality) so i'd prefer to use some other example, eg just disabling _all is sufficient to demonstrate this technique.

fforbeck commented:
Hi @clintongormley 
Yep, I got your point. 
I will update the docs and remove the `attachment` stuff.

Thanks for the review. :+1:

fforbeck commented:
Hi @clintongormley 
I have removed the `attachment` plugin details and added an example using the `_all` property, as you had recommended.

Please, take a look if possible.

Thank you!!

clintongormley commented:
Thanks @fforbeck - there were a couple of issues with your example which I've fixed, and i've simplified the text to just demonstrate the basics.

thanks for the PR

fforbeck commented:
@clintongormley 
Alright, no problem! Much better now :+1: 

Thanks
